src/docs/resources/architectural-overview.md

```diff
@@ -0,0 +1,966 @@
+
+Whenever you mutate a `State` object (for example, by incrementing the counter),
+you must call
+[`setState()`](https://api.flutter.dev/flutter/widgets/State/setState.html) to signal the framework to update the user interface by calling the `State`’s build method again.
+
+Having separate state and widget objects lets other widgets treat both stateless
+and stateful widgets in exactly the same way, without being concerned about
+losing state. Instead of needing to hold on to a child to preserve its state,
+the parent can create a new instance of the child at any time without losing the
+child’s persistent state. The framework does all the work of finding and reusing
+existing state objects when appropriate.
+
+### State management
+
+So, if many widgets can contain state, how is state managed and passed around
+the system?
+
+As with any other class, you can use a constructor in a widget to initialize its
+data, so a `build()` method can ensure that any child widget is instantiated
+with the data it needs:
+
+```dart
+@override
+Widget build(BuildContext context) {
+   return ContentWidget(importantState);
+}
+```
+
+As widget trees get deeper, however, passing state information up and down the
+tree hierarchy becomes cumbersome. So, a third widget type,
+[`InheritedWidget`](https://api.flutter.dev/flutter/widgets/InheritedWidget-class.html), provides an easy way to grab data from a shared ancestor. You can use `InheritedWidget` to create a state widget that wraps a common
+ancestor in the widget tree, as shown in this example:
+
+![Inherited
+widgets](../../images/arch-overview/inherited-widget.svg){:width="100%"}
+
+Whenever one of the `ExamWidget` or `GradeWidget` objects needs data from
+`StudentState`, it can now access it with a command such as:
+
+```dart
+final studentState = StudentState.of(context);
+```
+
+The `of(context)` call takes the build context (a handle to the current widget
+location), and returns [the nearest ancestor in the
+tree](https://master-api.flutter.dev/flutter/widgets/BuildContext/dependOnInheritedWidgetOfExactType.html)
+that matches the `StudentState` type. `InheritedWidget`s also offer an
+`updateShouldNotify()` method, which Flutter calls to determine whether a state
+change should trigger a rebuild of child widgets that use it.
+
+Flutter itself uses `InheritedWidget` extensively as part of the framework for
+shared state, such as the application’s _visual theme_, which includes
+[properties like color and type
+styles](https://api.flutter.dev/flutter/material/ThemeData-class.html) that are
+pervasive throughout an application. The `MaterialApp` `build()` method inserts
+a theme in the tree when it builds, and then deeper in the hierarchy a widget
+can use the `.of()` method to look up the relevant theme data, for example:
+
```

Comment:
Travis runs the latest stable release (in this case, 1.20) of the analyzer over the code. If the code is a snippet, like this, it will fail analysis. To fix that, add the following line just before the tick-tick-tick:

```
<!-- skip -->
```